### ISSUE TITLE
False positive with single expression functions that use multiple `return` keywords

### ISSUE THREAD

**Issue by Skeletonxf** [saveourtool/diktat#1588]
## Describe the bug

Diktat rewrites the following code:

```kotlin
data class Demo(val value: String) {
    companion object {
        fun from(string: String?): Demo? {
            return Demo(string ?: return null)
        }
    }
}
```

into

```kotlin
data class Demo(val value: String) {
    companion object {
        fun from(string: String?): Demo? = Demo(string ?: return null)
    }
}
```

## Expected behavior

Ditkat does not rewrite the above code, because it is not valid Kotlin to use the return keyword when a function is a single expression. The single expression function body is necessary to support the null input case.

## Observed behavior

Diktat rewrites the code causing a compilation error

## Steps to Reproduce

Write any function body which consists of a single expression but uses patterns like `?: return foo` which would not be supported as a single expression function

Adding any comment inside the function body stops diktat from rewriting the code, but this is not ideal since there is not always a need for such a comment.

## Environment information
* diktat version: 1.2.1
* build tool (maven/gradle): gradle
* how is diktat run (CLI, plugin, etc.): spotless (spotlessApply task)
* kotlin version: 1.7.10
* operating system: macOS
* link to a project (if your project is public): NA

**Comment by nulls** [comment]
hi @Skeletonxf,

Thanks for reporting the issue.
We will look into it
